src/containers/CreatePipelineRun/YAMLEditor.js

```diff
@@ -16,29 +16,41 @@ import {
   Button,
   Form,
   FormGroup,
-  InlineNotification
+  InlineNotification,
+  Loading
 } from 'carbon-components-react';
 import yaml from 'js-yaml';
-import React, { useState } from 'react';
+import React, { useEffect, useState } from 'react';
 import CodeMirror from '@uiw/react-codemirror';
 import { StreamLanguage } from '@codemirror/language';
 import { yaml as codeMirrorYAML } from '@codemirror/legacy-modes/mode/yaml';
 import { useNavigate } from 'react-router-dom-v5-compat';
 import { createPipelineRunRaw, useSelectedNamespace } from '../../api';
 
-export function CreateYAMLEditor({ code: initialCode = '' }) {
+export function CreateYAMLEditor({
+  code: initialCode,
+  loading = false,
+  loadingMessage = ''
+}) {
   const intl = useIntl();
   const navigate = useNavigate();
   const { selectedNamespace } = useSelectedNamespace();
 
-  const [{ code, isCreating, submitError, validationErrorMessage }, setState] =
+  const [{ isCreating, submitError, validationErrorMessage, code }, setState] =
     useState({
-      code: initialCode,
       isCreating: false,
       submitError: '',
-      validationErrorMessage: ''
+      validationErrorMessage: '',
+      code: initialCode
     });
 
+  useEffect(() => {
+    setState(state => ({
```

Comment:
we only want to do this `if(!loading)` right?

what happens in case of an error loading the PipelineRun?

Reply:
yeah, added

- in case of the error code mirror content will be empty, but editable (loading true -> false)
- no information about the error will be displayed to user
![image](https://user-images.githubusercontent.com/9289172/212793327-03304331-0914-4e47-b76e-501c388f8d33.png)

Reply:
OK that will do for now but we may revisit later to display some error message to the user. They can always refresh the page to retry if needed and it should be rare enough unless they're following an old link that was shared / bookmarked.